### PR TITLE
PR to add additional plots

### PR DESCRIPTION
**Pull request by mmeier93**
## This PR does the following:
It adds the ability to make additional plots to those produced by PRSice. It produces:

- For each covariate used in PRSice and for the for the best fitting PRS calculated by PRSice, a plot showing the PRS score vs the covariate
- A density plot showing, for the best fitting PRS calculated by PRSice, the distribution of PRS scores of cases vs controls

## Details of changes:

- 2 R scripts were made in order to make each of the 2 types of plots
- Those 2 Rscripts are used within a single new Nextflow process called `additional_plots`
- The environment.yml (and Dockerfile) had to be updated to include `snakecase` and `sm`, 2 R packages
- Since the PRScice plots are produced by the Nextflow process `polygen_risk_calcs` and the new plots by `additional_plots`, I had to make a channel to combine all plots before rending the markdown.

## Comments:

- For the moment, the markdown report contains hardcoded names of plots. This will need to be changed extensively to allow for a more dynamic type of report.
- The CloudOS tests are producing some inconsistancies when it comes to the exact results being produced compared to my local test (see below). 
- Issues have been added to the Issues section of this pipeline.

## My tests in Lifebit CloudOS

My first test (today, 07/10/20) works: https://cloudos.lifebit.ai/app/jobs/5f7d9fd1ff510d011360d02a
**However**:
- It produces different results (plots and PRSice outputs) to those I had locally
- It does however, seem to reproduce the results of last's week's sprint (Sprint #112): https://cloudos.lifebit.ai/app/jobs/5f7618d0ff510d01135c625f

**Importantly**, following our discussions, I did more looking around and I think I have found why my results did not replicate. It turns out I did not use the same input SAIGE summary statistics between my local desktop and CloudOS. I used the same for both CloudOS jobs so they replicate - looking at the file called `PRSice.log` is very helpful here. Indeed, although I don't have a record of the SAIGE input downloaded to my laptop (since it was done manually), the files in my laptop say: 

Laptop:
Screenshot 2020-10-07 at 20 58 20

CloudOS jobs:
Screenshot 2020-10-07 at 20 59 03

Final test, where I manually overrid the arguments specified in nextflow.config gave same results as the first 2 job:
https://cloudos.lifebit.ai/public/jobs/5f7e240f8f81710113021eb9

So it seems it must be been all due to different SAIGE inputs between my laptop and CloudOS jobs.

### CODE REVIEW

**Comment by cgpu** [comment]
> Importantly, following our discussions, I did more looking around and I think I have found why my results did not replicate. It turns out I did **not use the same input SAIGE summary statistics** between my local desktop and CloudOS. I used the same for both CloudOS jobs so they replicate - looking at the file called PRSice.log is very helpful here. Indeed, although I don't have a record of the SAIGE input downloaded to my laptop (since it was done manually), the files in my laptop say:

Nice detective work there @mmeier93!

File: assets/metadata_phenotypes_mapping_file.csv [hunk 1]
```diff
@@ -0,0 +1,568 @@
+FieldID Category Path,Level 2,FieldID,Array,Instances,Field Name,valueType,Sorting,FieldID Type,FieldID Type Units,Coding,Description Text,Description Participants No,Link,Description Stability,Description Category ID,Description ItemType,Description Strata,Description Sexed,order_phenotype,instance0_name,instance1_name,instance2_name,instance3_name,instance4_name,instance5_name,instance6_name,instance7_name,instance8_name,instance9_name,instance10_name,instance11_name,instance12_name,instance13_name,instance14_name,instance15_name,instance16_name,instance17_name,bucket_300,bucket_500,bucket_1000,bucket_2500,bucket_5000,bucket_10000,Original_name
```
Comment (cgpu) [inline assets/metadata_phenotypes_mapping_file.csv:1]: Task for @cgpu, upload to s3.

File: bin/plot_prs_vs_cov.R [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+plot_data <- inner_join(cov, prs, by = c( "iid" = "iid", "fid" = "fid"))
+plot_data <- suppressWarnings(plot_data %>% select(iid, prs, one_of(cov_to_keep)))
+
+#### Plot ####
+
+plot_data <- as.data.frame(plot_data)
+
+suppressWarnings(
+for (i in colnames(plot_data[-c(1,2)])) {
```
Comment (cgpu) [inline bin/plot_prs_vs_cov.R:43]: Here we could change the positional reference to the columns with the column names explicitly for readability?

File: bin/plot_prs_vs_density.R [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+#### Transform data ####
+
+colnames(pheno) <- snakecase::to_snake_case(colnames(pheno))
+colnames(prs) <- snakecase::to_snake_case(colnames(prs))
+plot_data <- inner_join(pheno, prs, by = c( "iid" = "iid", "fid" = "fid"))
+
+#### Plot density ####
+
+png("prs_vs_density.png")
```
Comment (cgpu) [inline bin/plot_prs_vs_density.R:47]: we could add an extra label here, to allow to the user to customise the output name:

```Suggestion
png(paste0(output_label, "prs_vs_density.png"))
```

File: environment.yml [hunk 4]
```diff
@@ -13,6 +13,11 @@ dependencies:
   - conda-forge::r-flexdashboard=0.5.1.1
   - conda-forge::r-stringi=1.4.3
   - conda-forge::r-tidyverse=1.3.0 
+<<<<<<< HEAD:environment.yml
+  - conda-forge::r-snakecase=0.11.0
+  - conda-forge::r-sm=2.2_5.6
+=======
+>>>>>>> origin/master:docker/environment.yml
```
Comment (cgpu) [inline environment.yml:20]:
```Suggestion
  - conda-forge::r-snakecase=0.11.0
  - conda-forge::r-sm=2.2_5.6
```

**Review by cgpu — APPROVED** [review]
Good to go @mmeier93 👍 !

File: bin/plot_prs_vs_density.R (hunk 3 above)
Comment (mmeier93) [reply bin/plot_prs_vs_density.R:47]: Ok I opened up an issue! 👍

**Comment by mmeier93** [comment]
> Good to go @mmeier93 👍 !

Thanks!! 👍

**Pull request merged**